ibm/service/power/resource_ibm_pi_volumes_detach.go

```diff
@@ -0,0 +1,110 @@
+		DeleteContext: resourceIBMPiVolumesDetachDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			Arg_CloudInstanceID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Cloud Instance ID of a PCloud Instance.",
+			},
+			Arg_InstanceID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "PCloud PVM Instance ID.",
+			},
+			Arg_DetachAllVolumes: {
+				Type:          schema.TypeBool,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{Arg_VolumeIDs},
+				Description:   "Indicates if all volumes, except primary boot volume, attached to the PVMInstance should be detached (default=false); required if volumeIDs is not provided.",
+			},
+			Arg_DetachPrimaryBootVolume: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Indicates if primary boot volume attached to the PVMInstance should be detached (default=false).",
+			},
+			Arg_VolumeIDs: {
+				Type:          schema.TypeList,
+				Optional:      true,
+				ForceNew:      true,
+				Description:   "List of volumes to be detached from a PVM instance; required if detachAllVolumes is not provided.",
+				Elem:          &schema.Schema{Type: schema.TypeString},
+				ConflictsWith: []string{Arg_DetachAllVolumes},
+			},
+		},
+	}
+}
+
+func resourceIBMPiVolumesDetachCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	pvmInstanceID := d.Get(Arg_InstanceID).(string)
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	body := &models.VolumesDetach{}
+	var volumeIDs []string
+	if _, ok := d.GetOk(Arg_VolumeIDs); ok {
+		for _, v := range d.Get(Arg_VolumeIDs).([]interface{}) {
+			volumeIDsItem := v.(string)
+			volumeIDs = append(volumeIDs, volumeIDsItem)
+		}
+		body.VolumeIDs = volumeIDs
+	}
+
+	if _, ok := d.GetOk(Arg_DetachAllVolumes); ok {
+		body.DetachAllVolumes = flex.PtrToBool(d.Get(Arg_DetachAllVolumes).(bool))
+	}
+	if _, ok := d.GetOk(Arg_DetachPrimaryBootVolume); ok {
+		body.DetachPrimaryBootVolume = flex.PtrToBool(d.Get(Arg_DetachPrimaryBootVolume).(bool))
+	}
+
+	volClient := st.NewIBMPIVolumeClient(ctx, sess, cloudInstanceID)
+	volinfo, err := volClient.BulkVolumeDetach(pvmInstanceID, body)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[DEBUG] Volumes detach accepted:  %s", *volinfo.Summary)
+
+	d.SetId(fmt.Sprintf("%s/%s", cloudInstanceID, pvmInstanceID))
+
+	return resourceIBMPiVolumesAttachRead(ctx, d, meta)
+}
+
+func resourceIBMPiVolumesDetachRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	return nil
+}
+
+func resourceIBMPiVolumesDetachDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
```

Comment:
I was looking through some of the other resources that have a stub for read and delete. I saw that at least one of them set 
```
d.SetId("")
```
in the delete before returning nil.

Do we need to do that here?

Reply:
I have added that